tasks/export_abi.js

```diff
@@ -24,10 +24,15 @@ task('export-abi', async function (args, hre) {
 
     if (!abi.length) continue;
 
+    let outputName = contractName;
+    if (typeof config.rename === 'function') {
+      outputName = config.rename(fullName);
+    }
+
     const destination = path.resolve(
       outputDirectory,
       config.flat ? '' : sourceName,
-      contractName
+      outputName
```

Comment:
```suggestion
      config.rename?.(fullName) || contractName
```

I'd probably do it like this, and remove the code above.  But still wondering whether the `rename` function needs to account for `sourceName`.

Reply:
I think it would be fine to pass `sourceName` and `contractName` instead of the fully qualified name. I wasn't sure which was better.

Reply:
```suggestion
      config.rename?.(sourceName, contractName) || contractName
```
Thoughts on this?

Reply:
Not very compatible with the `flat` option.  Maybe throw an error if both `flat` and `rename` are used?  Or replace `flat` with a hardcoded `rename` option like `rename: 'flat'`?

Reply:
I like that idea. Maybe `full`, `flat`, of a function?

Reply:
Implemented!